tests/test_async_http_client.py

```diff
@@ -0,0 +1,139 @@
+from platform import python_version
+
+import httpx
+import pytest
+from unittest.mock import AsyncMock
+
+from workos.utils.http_client import AsyncHTTPClient
+
+
+class TestAsyncHTTPClient(object):
+    @pytest.fixture(autouse=True)
+    def setup(self):
+        def handler(request: httpx.Request) -> httpx.Response:
+            return httpx.Response(200, json={"message": "Success!"})
+
+        self.http_client = AsyncHTTPClient(
+            base_url="https://api.workos.test",
+            version="test",
+            transport=httpx.MockTransport(handler),
+        )
+
+        self.http_client._client.request = AsyncMock(
+            return_value=httpx.Response(200, json={"message": "Success!"}),
+        )
+
+    @pytest.mark.asyncio
+    async def test_get_request(self):
+        response = await self.http_client.request(
+            "events", method="GET", params={"test_param": "test_value"}, token="test"
```

Comment:
`token` gets put in the `Authorization` header. We could probably simplify it a bit an have it part of the HTTP client logic, but kept it as-is for now since that's how it's been passed in the past.